src/guidance/guidance_processing.cpp

```diff
@@ -0,0 +1,325 @@
+#include "guidance/guidance_processing.hpp"
+#include "guidance/turn_analysis.hpp"
+#include "guidance/turn_lane_handler.hpp"
+
+#include "extractor/intersection/intersection_analysis.hpp"
+
+#include "util/assert.hpp"
+#include "util/percent.hpp"
+
+#include <tbb/blocked_range.h>
+#include <tbb/pipeline.h>
+#include <tbb/task_scheduler_init.h>
+
+namespace osrm
+{
+namespace guidance
+{
+
+void processGuidanceTurns(const util::NodeBasedDynamicGraph &node_based_graph,
```

Comment:
Nitpick on naming here: What we are doing is annotation the turn with guidance information, so for example something like `guidance::annotateTurns` might be sufficient as a name.